persim/images.py

```diff
@@ -210,3 +220,695 @@ def show(self, imgs, ax=None):
+                dim1xs = np.outer(dim1, xs)
+                dim1rs = np.outer(dim1, rs)
+                dim1w = np.outer(dim1, w)
+                rhk16dim2 = np.outer(rhk16, dim2)
+                hkdim2 = np.outer(hk, dim2)
+                asr1 = -1.0 * (np.divide(xmy2dim2, dim1xs) + hkdim2) / 2.0
+
+                ind1 = asr1 > -100
+                cdim2 = np.outer(rhk8, dim2)
+                sp1 = 1.0 + np.multiply(np.multiply(cdim2, dim1xs), 1.0 + np.multiply(rhk16dim2, dim1xs))
+                ep1 = np.divide(np.exp(np.divide(-np.multiply(hkdim2, (1.0 - dim1rs)),
+                                                 2.0 * (1.0 + dim1rs))), dim1rs)
+                bvn = bvn + np.sum(np.multiply(np.multiply(np.multiply(sopmr, dim1w), np.exp(np.multiply(asr1, ind1))),
+                                               np.multiply(ep1, ind1) - np.multiply(sp1, ind1)), axis=1)
+            bvn = -bvn / (2.0 * np.pi)
+
+        if r > 0:
+            bvn = bvn + _norm_cdf(-np.maximum(dh, dk))
+        elif r < 0:
+            bvn = -bvn + np.maximum(0, _norm_cdf(-dh) - _norm_cdf(-dk))
+
+    return bvn
+
+
+def _gauss_legendre_quad(r):
+    """
+    Return weights and abscissae for the Legendre-Gauss quadrature integral approximation
+    :param r: correlation
+    :return:
+    """
+    if np.abs(r) < 0.3:
+        lg = 3
+        w = np.array([0.1713244923791705, 0.3607615730481384, 0.4679139345726904])
+        x = np.array([0.9324695142031522, 0.6612093864662647, 0.2386191860831970])
+    elif np.abs(r) < 0.75:
+        lg = 6
+        w = np.array([.04717533638651177, 0.1069393259953183, 0.1600783285433464,
+                      0.2031674267230659, 0.2334925365383547, 0.2491470458134029])
+        x = np.array([0.9815606342467191, 0.9041172563704750, 0.7699026741943050,
+                      0.5873179542866171, 0.3678314989981802, 0.1252334085114692])
+    else:
+        lg = 10
+        w = np.array([0.01761400713915212, 0.04060142980038694, 0.06267204833410906,
+                      0.08327674157670475, 0.1019301198172404, 0.1181945319615184,
+                      0.1316886384491766, 0.1420961093183821, 0.1491729864726037,
+                      0.1527533871307259])
+        x = np.array([0.9931285991850949, 0.9639719272779138, 0.9122344282513259,
+                      0.8391169718222188, 0.7463319064601508, 0.6360536807265150,
+                      0.5108670019508271, 0.3737060887154196, 0.2277858511416451,
+                      0.07652652113349733])
+
+    return lg, w, x
+
+
+"""
+Weight functions:
+
+A valid weight is a python function of the form weight(birth, persistence, **kwargs) defining a scalar-valued function over the birth-persistence plane, where birth and persistence are numpy arrays of equal length. To ensure stability, functions should vanish continuously at the
+line persistence = 0 (see [PI] for details).
+"""
+
+def linear_ramp(birth, pers, low=0.0, high=1.0, start=0.0, end=1.0):
```

Comment:
Same with these weight functions, moved to a separate file.

Reply:
I moved these to images_weights.py.